File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/cache/EntityCacheTest.java
package com.coupang.springframework.data.requery.java.cache;

import com.coupang.springframework.data.requery.java.domain.AbstractDomainTest;
import com.coupang.springframework.data.requery.java.domain.basic.BasicUser;
import io.requery.EntityCache;
import io.requery.cache.EntityCacheBuilder;
import io.requery.meta.EntityModel;
import org.junit.Before;
import org.junit.Test;

import javax.cache.CacheManager;
import javax.cache.Caching;
import javax.cache.spi.CachingProvider;
import javax.inject.Inject;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * EntityCacheTest
 *
 * @author debop@coupang.com
 * @since 18. 6. 5
 */
public class EntityCacheTest extends AbstractDomainTest {

    @Inject
    EntityModel entityModel;

    private EntityCache cache;

    private EntityCache getCache(EntityModel entityModel) {
        CachingProvider provider = Caching.getCachingProvider();
        CacheManager cacheManager = provider.getCacheManager();

        return new EntityCacheBuilder(entityModel)
            .useReferenceCache(false)
            .useSerializableCache(true)
            .useCacheManager(cacheManager)
            .build();
    }

    @Before
    public void setup() {
        if (cache == null) {
            cache = getCache(entityModel);
        }
    }

    @Test
    public void serializeEntityToCache() {
        BasicUser user = new BasicUser();
        int id = 100;

        cache.put(BasicUser.class, id, user);

        BasicUser loaded = cache.get(BasicUser.class, id);

        assertThat(loaded).isNotNull();
        assertThat(loaded).isEqualTo(user);
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/cache/EntityCacheTest.kt
package com.coupang.springframework.data.requery.cache

import com.coupang.springframework.data.requery.domain.AbstractDomainTest
import com.coupang.springframework.data.requery.domain.Models
import com.coupang.springframework.data.requery.domain.functional.FuncPerson
import com.coupang.springframework.data.requery.domain.functional.RandomData
import io.requery.EntityCache
import io.requery.cache.EntityCacheBuilder
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import javax.cache.Caching


/**
 * com.coupang.springframework.data.requery.cache.EntityCacheTest
 * @author debop
 * @since 18. 6. 2
 */
class EntityCacheTest: AbstractDomainTest() {

    private val cache: EntityCache by lazy {
        val provider = Caching.getCachingProvider()
        val cacheManager = provider.cacheManager

        EntityCacheBuilder(Models.DEFAULT)
            .useReferenceCache(false)
            .useSerializableCache(true)
            .useCacheManager(cacheManager)
            .build()
    }

    @Test
    fun `serialize entity and get put`() {

        val p = RandomData.randomPerson()

        val id = 100
        cache.put(FuncPerson::class.java, id, p)

        val d = cache.get(FuncPerson::class.java, id)

        assertThat(d).isNotNull
        assertThat(d === p).isFalse()
        assertThat(d).isEqualTo(p)
    }
}